pallets/redeem/src/lib.rs

```diff
@@ -917,7 +931,12 @@ impl<T: Config> Pallet<T> {
 
 		ensure!(redeem.vault == vault_id, Error::<T>::UnauthorizedVault);
 
-		let reimbursed_amount = redeem.amount().checked_add(&redeem.transfer_fee())?;
+		//let reimbursed_amount = redeem.amount().checked_add(&redeem.transfer_fee())?;
+
+		let Some((amount,currency_id)) = CancelledRedeemAmount::<T>::take(redeem_id) else {
+			return Err(DispatchError::from(Error::<T>::RedeemIdNotFound));
+		};
+		let reimbursed_amount = Amount::new(amount, currency_id);
```

Comment:
To follow up on the other ticket, here I would add a check to see if an item for the `redeem_id` exists in the map or not. And based on that we determine the reimbursed amount.

```suggestion
                 // Here, we check if the `transfer_funds_saturated()` function was used during cancellation and if that's not the case, calculate the reimbursed_amount
		let reimbursed_amount = match CancelledRedeemAmount::<T>::take(redeem_id) {
		Some(saturated_amount) => saturated_amount,
		None => {
                      redeem.amount().checked_add(&redeem.transfer_fee())?
		};
```

WDYT @b-yap?

Reply:
changed it